packages/@eventual/core-runtime/src/handlers/entity-stream-worker.ts

```diff
@@ -18,19 +24,44 @@ export function createEntityStreamWorker(
 ): EntityStreamWorker {
   registerWorkerIntrinsics(dependencies);
 
-  return async (item) =>
+  return async (entityName, streamName, items) =>
     serviceTypeScope(ServiceType.EntityStreamWorker, async () => {
       const streamHandler = entities()
-        .get(item.entityName)
-        ?.streams.find((s) => s.name === item.streamName);
+        .get(entityName)
+        ?.streams.find((s) => s.name === streamName);
+
       if (!streamHandler) {
-        throw new Error(`Stream handler ${item.streamName} does not exist`);
+        throw new Error(`Stream handler ${streamName} does not exist`);
       }
-      return await streamHandler.handler(item, {
+
+      const context: EntityStreamContext = {
+        stream: { entityName, streamName },
         service: {
           serviceName: getLazy(dependencies.serviceName),
           serviceUrl: getLazy(dependencies.serviceUrl),
         },
-      });
+      };
+
+      if (streamHandler.kind === "EntityBatchStream") {
+        const result = await streamHandler.handler(items, context);
+
+        return { failedItemIds: result?.failedItemIds ?? [] };
+      } else {
+        const results = await promiseAllSettledPartitioned(
+          items,
+          async (item) => {
+            return await streamHandler.handler(item, context);
+          }
+        );
+
+        return {
+          failedItemIds: [
+            ...results.rejected.map(([e]) => e.id),
+            ...results.fulfilled
+              .filter(([, r]) => r === false)
+              .map(([e]) => e.id),
+          ],
+        };
+      }
```

Comment:
Cool, so this supports partial successful items?

Reply:
Yeah!

The single item handler you can return false or throw for a partial failure (this was there before).

The batch API support returning the IDs to "fail".